### PR TITLE
Documentation - Update JavaScript for search pagination 

### PR DESCRIPTION
**Pull request by akshay-gupta7**
In response to issue https://github.com/wagtail/wagtail/issues/9102 :

Inline JavaScript in https://github.com/wagtail/wagtail/blob/main/docs/_templates/search.html has been moved to separate external JS file
onclick events has been replaced with addEventListener events
Now one button per page is displayed, Page 2 of 7, along with a previous and a Next button

### CODE REVIEW

File: docs/_templates/search.html
Comment (lb-) [inline docs/_templates/search.html:20]: if `pagination-previous` and `pagination-next` are still acting as buttons (from what I can see they are), they should still be buttons in the DOM.

```suggestion
        <button class="pagination-next" id="pagination-next" type="button" data-pagination-next></button>
```
also, while we are changing this, it might be nicer to pick up the JS section via a data attribute `data-pagination-next` as per our current UI guidelines.

https://docs.wagtail.org/en/latest/contributing/ui_guidelines.html#html-guidelines

File: docs/_templates/search.html
Comment (lb-) [inline docs/_templates/search.html:19]: Unless there is a reason otherwise, we should lean towards semantic elements, `ul` and `li` for the inner items make sense here.

File: docs/_templates/search.html
Comment (lb-) [inline docs/_templates/search.html:27]: note - it appears your editor has removed a line break at the end of the file, please can you re-add it.

File: docs/_static/js/search.js
Comment (lb-) [inline docs/_static/js/search.js:7]: it would be best to resolve these linting errors a different way.

* First, try to refactor the JS to avoid them (no-use-before-define) should be easy, same with no-underscore-dangle and no-undef
* for others, try to put the eslint ignore in the actual line of the change (e.g. for the console, I think there is only one line)
* This helps us ensure that new code added to this file adheres to the eslint styleguide
* I know you copied/pasted this code and it simply was not linted before, but we should try to have a balance of fixing things that are important for code readability/bugs when bring to its own file.

File: docs/_static/js/search.js
Comment (lb-) [inline docs/_static/js/search.js:123]: no need to add this extra bit to the comment here 

```suggestion
    // Register event listeners
```

File: docs/_static/js/search.js
Comment (lb-) [inline docs/_static/js/search.js:117]: It would be good to bring all labels up to a const/variable in the upper scope of each function, easier to use.

Ideally however, we should keep the label in the html and then just hide/show the button when not needed.

**Review by lb- — CHANGES_REQUESTED** [review]
@akshay-gupta7 thanks so much for taking the time to contribute!

This is a great start, I have some feedback which is mostly accessibility/coding guidelines related - please have a go at fixing these up and add a message on the PR when ready to review again.

I have not run this locally yet, mostly just feedback on the code itself, not the functionality.

File: docs/_static/js/search.js
Comment (akshay-gupta7) [reply docs/_static/js/search.js]: @lb- , refactored JavaScript files to remove most of the linting errors. I even removed inline JS code from layout.html and merged it with search.js.
Added //eslint-disable to only unavoidable specific Lines of Code.

File: docs/_static/js/search.js
Comment (akshay-gupta7) [reply docs/_static/js/search.js]: @lb- , moved labels to the upper scope of the function

File: docs/_templates/search.html
Comment (akshay-gupta7) [reply docs/_templates/search.html:26]: Couldn't find the line break. Apologies in case I am missing something.

File: docs/_templates/search.html
Comment (akshay-gupta7) [reply docs/_templates/search.html]: Added data-attributes to the buttons

File: docs/_templates/search.html
Comment (lb-) [reply docs/_templates/search.html:26]: Ok maybe the line break was there but has been removed. Not a problem for this PR right now.

**Review by lb- — COMMENTED** [review]
Just a few more tweaks but also still trying to see if there is a simpler way to get the dynamic buttons working without too much Dom element creation. 

On the surface the different is small but it makes understanding what the code is doing easier and hopefully avoids lots of manual handing of adding basics like classes and content.

File: docs/_templates/search.html
Comment (akshay-gupta7) [reply docs/_templates/search.html:18]: Hi @lb- , The reason I was forced to do create a Next and Previous button dynamically is because I replaced onclick with AddEventListerners and the feature of this is that it gets triggered for all the times it was mentioned in the DOM tree, whereas onclick gets triggered only for the last one. Hence, I created a Next/Previous button and dynamically replaced it on every page, so that the Listener event gets triggered only once.
Let me know if I was able to explain this, and also if you have any suggestions to circumvent this problem.

Comment (lb-) [reply docs/_templates/search.html:18]: Hmm. Good explanation - I'll have a better look tomorrow and see if we can find a simpler way. 

Thanks for the response and your persistence on getting this through.

Comment (akshay-gupta7) [reply docs/_templates/search.html:18]: Thank you so much. Meanwhile, I'll also keep at it to see if I find a solution.

File: docs/_static/js/search.js
Comment (akshay-gupta7) [reply docs/_static/js/search.js]: Worked like a charm, thanks!

File: docs/_templates/search.html
Comment (akshay-gupta7) [reply docs/_templates/search.html]: @lb- , managed to solve this roadblock using removeEventListener functions. Also, I had to take some operations in new functions, to follow the correct format of using these remove event functions. Please do have a look at the code and let me know your thoughts.
Thanks!

**Comment by akshay-gupta7** [comment]
@lb-, did you get a chance to review this. Hopefully this is working now.

**Comment by lb-** [comment]
> @lb-, did you get a chance to review this. Hopefully this is working now.

Thanks @akshay-gupta7  - I have flagged myself for another review - will get to this soon.

**Review by lb- — CHANGES_REQUESTED** [review]
@akshay-gupta7 thanks again for putting this together. 

However I think this still needs a solid amount of work before we can merge this in. Please don't be discouraged, this is a solid effort. 

Here's a bit of a plan for this - assuming you're still keen to take this on 
(I'll put this notes on the main issue with some code snippets also)

Maybe each of these steps could be a seperate PR and this original PR gets moved to draft. 

1. Move the JS to search.js - just doing this and solving some of the linting rules (which have changed slightly since this PR was created). Also check for console errors on non-search results pages. 
2. Moving the existing onClick handlers to event listeners - without changing any of the DOM structure.
3. Potential just moving to data attributes to 'select' elements instead of IDs.
4. Now tackle the re-work of the content. There are a few issues with the current solution in this code - mainly that it's a bit too complex for what we need and the 'page X of Y' label does not need to be a button anymore. Just a label. I'll put some notes on the main issue about this. 

I actually think the page content changes is probably better suited to its own issue. 

But again - good on you for giving this a solid go. But for now it's best we tackle.and review this in smaller steps.

**Comment by akshay-gupta7** [comment]
@lb- , thank you for taking out the time to review this. I am still keen on working through this issue.
I checked your notes and code snippets and I agree with you that we should make the existing code simpler and in accordance with the best practices.
I will tackle this issue in separate PRs to ensure we have nailed all the smaller steps.

File: docs/_templates/search.html
Comment (Aym-n) [reply docs/_templates/search.html]: @lb- Can't Find Search.js File Mentioned In This PR

**Comment by Aym-n** [comment]
- [ ] Move the JS to search.js
- [ ] Moving the existing onClick handlers to event listeners 
- [ ] Potential just moving to data attributes to 'select' elements instead of IDs.

**Comment by Aym-n** [comment]
All the JS in `docs\_templates\search.html' has already been transferred to the `docs\search.js`

**Comment by Aym-n** [comment]
All the `onClick` handlers have already been changed to `Event Listeners`

**Comment by Aym-n** [comment]
made changes to this PR in PR#9941

**Comment by thibaudcolas** [comment]
Closing in favour of #9941 which seems to be a bit further ahead. Thank you for giving this a go @akshay-gupta7! Your work is much appreciated so we’ll make sure to credit you as well when #9941 gets merged.

**Pull request closed without merging**
